extractor/src/main/java/org/schabi/newpipe/extractor/stream/StreamExtractor.java

```diff
@@ -349,4 +350,72 @@ protected long getTimestampSeconds(String regexPattern) throws ParsingException
             return 0;
         }
     }
+
+    /**
+     * The host of the stream (Eg. peertube.cpy.re).
+     * If the host is not available, or if the service doesn't use
+     * a federated system, but a centralised system,
+     * you can simply return an empty string.
+     * @return the host of the stream or an empty String.
+     * @throws ParsingException
+     */
+    @Nonnull
+    public abstract String getHost() throws ParsingException;
+
+    /**
+     * The privacy of the stream (Eg. Public, Private, Unlisted…).
+     * If the privacy is not available you can simply return an empty string.
+     * @return the privacy of the stream or an empty String.
+     * @throws ParsingException
+     */
+    @Nonnull
+    public abstract String getPrivacy() throws ParsingException;
+
+    /**
+     * The name of the category of the stream.
+     * If the category is not available you can simply return an empty string.
+     * @return the category of the stream or an empty String.
+     * @throws ParsingException
+     */
+    @Nonnull
+    public abstract String getCategory() throws ParsingException;
+
+    /**
+     * The name of the licence of the stream.
+     * If the licence is not available you can simply return an empty string.
+     * @return the licence of the stream or an empty String.
+     * @throws ParsingException
+     */
+    @Nonnull
+    public abstract String getLicence() throws ParsingException;
+
+    /**
+     * The language of the stream.
+     * If the language is not available you can simply return an empty string.
+     * @return the licence of the stream or an empty String.
+     * @throws ParsingException
+     */
+    @Nonnull
+    public abstract String getLanguageInfo() throws ParsingException;
+
+    /**
+     * The list of tags of the stream.
+     * If the tag list is not available you can simply return an empty list.
+     * @return the list of tags of the stream or an empty list.
+     * @throws ParsingException
+     */
+    @Nonnull
+    public abstract List<String> getTags() throws ParsingException;
+
+    /**
+     * The support information of the stream.
+     * see: https://framatube.org/videos/watch/ee408ec8-07cd-4e35-b884-fb681a4b9d37
+     * (support button).
+     * If the support information are not available,
+     * you can simply return an empty String.
+     * @return the support information of the stream or an empty String.
+     * @throws ParsingException
+     */
+    @Nonnull
+    public abstract String getSupportInfo() throws ParsingException;
```

Comment:
I think that theScrabi once tried to get this info from YouTube channels, too. Can you please look  whether there is a field in the channel extractor which provides support / donation info and alter the name if needed?

Reply:
The only thing I see is getDescription which provides channel's about page. YouTube doesn't have support info or whatever, people put in their video's description / tell about it in the video, outro…

Notice this method is for in video Extractor, video, because on PeerTube donation button is on videos, and if a user decide to put the support button on a video, it doesn't mean it will be in all videos. See https://framatube.org/videos/watch/c07db2e9-6247-480f-a87e-3ec09e67596c and https://framatube.org/videos/watch/a791c2ff-7588-4ddf-905c-5b7dde051ae8 

BTW, getDescription for YouTube is only giving first 160 characters.

I don't really understand what you want.

Reply:
On channel pages, there was an attempt to extract the links and buttons in the green box, but it seems the change was not merged. So my review is pointless then. 
![image](https://user-images.githubusercontent.com/17365767/73109277-ffa96f00-3f02-11ea-8f84-319aaad28c81.png)

> BTW, getDescription for YouTube is only giving first 160 characters.

I could not reproduce this.

Reply:
I was talking about channel getDescription
Steps to reproduce:
go in YoutubeChannelExtractorTest.java
replace any TestDescription() methods content by
String desc = extractor.getDescription();
System.out.println(desc.length());

output is always 160
BTW, line 281: //TODO: Description get cuts out, because the og:description is optimized and don't have all the content

but this is not related to the pr so i'm marking this conversation resolved